check_nextcloud.php

```diff
@@ -22,36 +23,60 @@ function convert_filesize($bytes, $decimals = 2) {
   return sprintf("%.{$decimals}f %s", $bytes / pow(1024, $factor), @$size[$factor]);
 }
 
-
-// get commands passed as arguments
-$options = getopt("H:U:u:p:s::");
-if (!is_array($options) ) {
-  print "There was a problem reading the passed option.\n\n";
-  exit(1);
-}
-
-if (count($options) < "4") {
+function show_helptext() {
   print "check_nextcloud.php - Monitoring plugin to check the status of nextcloud serverinfo app.\n
 You need to specify the following parameters:
   -H:  hostname of the nextcloud instance, e.g. cloud.example.com
   -U:  uri of the nextcloud serverinfo api, you'll find it at https://cloud.example.com/settings/admin/serverinfo
+  -T:  authenticate using serverinfo token (either -T or -u and -p)
   -u:  username to authenticate against the API endpoint
   -p:  password to authenticate against the API endpoint
   -s:  (optional) should the check be done over HTTPS? (default: true)  \n\n";
+}
+
+
+
+// get commands passed as arguments
+$options = getopt("H:U:u:p:T:s::");
+if (!is_array($options) ) {
+  print "There was a problem reading the passed option.\n\n";
+  exit(1);
+}
+
+if( !isset($options['H']) ) {
+  show_helptext();
   exit(2);
 }
 
 $nchost = trim($options['H']);
-$ncuri = trim($options['U']);
+$ncuri = isset($options['U']) ? trim($options['U']) : "/ocs/v2.php/apps/serverinfo/api/v1/info";
 $ncssl = (isset($options['s']) && is_bool($options['s'])) ? $options['s'] : true;
-$ncuser = trim($options['u']);
-$ncpass = trim($options['p']);
-$ncurl = ($ncssl ? "https://" : "http://") . $ncuser . ":" . $ncpass . "@" . $nchost . $ncuri;
+$ncsit =  isset($options['T']) ? trim($options['T']) : "";
```

Comment:
Can you give this variable a better name? I don't know what 'sit' is standing for. I assume server-info-token? Better would probably be `$nctoken`.

Reply:
ok